stanford_paragraph_types.module

```diff
@@ -1,7 +1,209 @@
 <?php
+
 /**
  * @file
  * Code for the Stanford Paragraph Types feature.
  */
 
 include_once 'stanford_paragraph_types.features.inc';
+
+/**
+ * Implements hook_theme_registry_alter().
+ *
+ * @see https://www.drupal.org/docs/7/creating-custom-modules/howtos/using-template-tplphp-files-in-your-own-module
+ */
+function stanford_paragraph_types_theme_registry_alter(&$theme_registry) {
+  $module_path = drupal_get_path('module', 'stanford_paragraph_types');
+  $template_objects = drupal_find_theme_templates($theme_registry, '.tpl.php', $module_path);
+  foreach ($template_objects as $key => $template_object) {
+    if (!isset($theme_registry[$key]['theme path']) || !preg_match('#/themes/#', $theme_registry[$key]['theme path'])) {
+      $theme_registry[$key]['theme path'] = $module_path;
+      $theme_registry[$key] = array_merge($theme_registry[$key], $template_object);
+      $theme_registry[$key]['type'] = 'module';
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_entity().
+ */
+function stanford_paragraph_types_preprocess_entity(&$vars) {
+  if (isset($vars['paragraphs_item'])) {
+    drupal_add_css(drupal_get_path('module', 'stanford_paragraph_types') . '/css/stanford_paragraph_types.css');
+    $entity = &$vars['paragraphs_item'];
+
+    // Adds to push following fields down.
+    $vars['classes_array'][] = 'clearfix';
+
+    switch ($entity->bundle) {
+      case 'p_hero':
+        // Adds class to tell where the overlay should live.
+        $classes = explode(' ', $entity->field_p_hero_alignment[LANGUAGE_NONE][0]['value']);
+        foreach ($classes as $class) {
+          $vars['classes_array'][] = $class;
+        }
+        break;
+
+      default:
+        // Other paragraph types that should be centered and not full width.
+        $vars['classes_array'][] = 'span12';
+        $vars['classes_array'][] = 'center';
+        break;
+    }
+  }
+
+  if ($vars['entity_type'] == 'field_collection_item' && isset($vars['field_p_card_image'])) {
+    if (!empty($vars['field_p_card_file']) && !empty($vars['field_p_card_cta'])) {
+      // If a file is added to the cards, we dont want to display the CTA.
+      unset($vars['content']['field_p_card_cta']);
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_block().
+ */
+function stanford_paragraph_types_preprocess_block(&$vars) {
+  if (isset($vars['elements']['#blockreference_element']) &&
+    $vars['elements']['#blockreference_element']['field']['field_name'] == 'field_p_menu_below_menu'
+  ) {
+
+    // Change span* classes so the block will be 100% of the field area for
+    // blockreference field rendered blocks.
+    array_walk($vars['classes_array'], function (&$array) {
```

Comment:
Nice. I like this. It is aggressive but I think this will solve more problems then it may create.